custom_components/sat/cycles.py

```
@@ -0,0 +1,532 @@
+    def cycles_last_hour(self) -> float:
+        """Cycles per hour, extrapolated from the cycle's window."""
+        now = self._current_time_hint()
+        if now is not None:
+            self._prune_cycles_window(now)
+
+        count = len(self._cycle_end_times_window)
+        return count * 3600.0 / self._cycles_window_seconds
+
+    @property
+    def duty_ratio_last_15m(self) -> float:
+        """Duty ratio (0.0–1.0) over the duty window, derived from recorded cycles."""
```

Ruff (0.14.10)

Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)